### PR TITLE
Wait for elements number

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java [hunk 1]
```diff
@@ -90,6 +93,45 @@ private <T> void waitForConditionValidatingVisibility(Locator locator,
...
+     * @param comparisonRule The rule to match the variable value. The supported rules:
+     *                       <ul>
+     *                       <li>less than (&lt;)</li>
+     *                       <li>less than or equal to (&lt;=)</li>
+     *                       <li>greater than (&gt;)</li>
+     *                       <li>greater than or equal to (&gt;=)</li>
+     *                       <li>equal to (=)</li>
+     *                       <li>not equal to (!=)</li>
+     *                       </ul>
+     * @param expectedNumberOfElements
```
Comment (uarlouski) [inline vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java:108]: missing description, also can we use the same param name across wording and method?

File: vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java [hunk 2]
```diff
@@ -90,6 +93,45 @@ private <T> void waitForConditionValidatingVisibility(Locator locator,
         waitForCondition(conditionFactory.apply(locator));
     }
 
+    /**
+     * Waits for expected number of elements.
+     * @param locator The locator to find an element.
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java:98]:
```suggestion
     * @param locator The locator to find elements.
```

File: docs/modules/plugins/partials/generic-ui-steps.adoc [hunk 3]
```diff
@@ -1,3 +1,30 @@
+=== Wait for expected elements number
+
+Waits for the expected number of elements located by locator.
+
+TIP: In order to wait for invisible elements one could rely on <<_visibility_types>>.
```
Comment (valfirst) [inline docs/modules/plugins/partials/generic-ui-steps.adoc:5]:
```suggestion
TIP: In order to wait for invisible elements one rely on <<_visibility_types>>.
```

File: vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java [hunk 4]
```diff
@@ -90,6 +93,45 @@ private <T> void waitForConditionValidatingVisibility(Locator locator,
...
+     * @param comparisonRule The rule to match the variable value. The supported rules:
+     *                       <ul>
+     *                       <li>less than (&lt;)</li>
+     *                       <li>less than or equal to (&lt;=)</li>
+     *                       <li>greater than (&gt;)</li>
+     *                       <li>greater than or equal to (&gt;=)</li>
+     *                       <li>equal to (=)</li>
+     *                       <li>not equal to (!=)</li>
+     *                       </ul>
+     * @param number         The expected number of elements
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java:108]:
```suggestion
     * @param number         The expected number of elements.
```

File: vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java [hunk 5]
```diff
@@ -90,6 +93,45 @@ private <T> void waitForConditionValidatingVisibility(Locator locator,
...
+                @Override
+                public Boolean apply(SearchContext searchContext)
+                {
+                    return rule.matches(searchActions.findElements(context, locator).size());
+                }
+
+                @Override
+                public String toString()
+                {
+                    return "number of elements located by: " + locator.toHumanReadableString() + " is "
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java:127]:
```suggestion
                    return "number of elements located by \"" + locator.toHumanReadableString() + "\" is "
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**